tests/acceptance/test_data_mappers.py

```diff
@@ -12,17 +12,21 @@
 
 
 @pytest.mark.auth
-def test_auth(api_client, data_mapper_base_endpoint):
+def test_auth(api_client, data_mapper_base_endpoint, stack):
     headers = {"Authorization": None}
+    status_code = 403 if stack["AuthMethod"] == "IAM" else 401
```

Comment:
I have moved the logic for this into a `fixture` so that the test no longer contains the condition. I know this is not truely what you meant. But first step.

Reply:
I have just split the marks on the tests so there are now separate cognito and iam versions.